src/chaincode.hpp

```diff
@@ -46,9 +46,9 @@ class ChainCode {
     void Serialize(uint8_t *buffer) const;
     std::vector<uint8_t> Serialize() const;
 
- private:
     // Prevent direct construction, use static constructor
```

Comment:
This statement is not longer true then
```suggestion
```

Reply:
That's true, but it's also true for the other constructors that have been made public by us. It's obviously a small diff, but generally should we be doing that? Or staying as close as possible to upstream

Reply:
> it's also true for the other constructors that have been made public by us

Hmm.. right.

I mean, at the end it doesn't matter lot and i will still give you an ACK if you leave it there 😅 its just.. for me it feels like the wrong way of thinking to "not fully complete changes" in favour of staying close to upstream. You would probably also not leave 10 lines of comments describing a method you removed or leave some unused variable declarations only because it would be closer to the upstream :D But thats probably personal preference and im pretty sure you did run into more merge conflicts yet which maybe lets you rather think like this hehe